src/main/java/net/minecraftforge/server/command/CommandSetDimension.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2017.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.server.command;
+
+import net.minecraft.command.CommandBase;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.command.WrongUsageException;
+import net.minecraft.entity.Entity;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.World;
+import net.minecraftforge.common.DimensionManager;
+import net.minecraftforge.common.util.ITeleporter;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
+public class CommandSetDimension extends CommandBase
+{
+    @Override
+    public String getName()
+    {
+        return "setdimension";
+    }
+
+    @Override
+    public List<String> getAliases()
+    {
+        return Collections.singletonList("setdim");
+    }
+
+    @Override
+    public String getUsage(ICommandSender sender)
+    {
+        return "commands.forge.setdim.usage";
+    }
+
+    @Override
+    public List<String> getTabCompletions(MinecraftServer server, ICommandSender sender, String[] args, @Nullable BlockPos targetPos)
+    {
+        if (args.length > 2 && args.length <= 5)
+        {
+            return getTabCompletionCoordinate(args, 2, targetPos);
+        }
+        return Collections.emptyList();
+    }
+
+    @Override
+    public int getRequiredPermissionLevel()
+    {
+        return 2;
+    }
+
+    @Override
+    public void execute(MinecraftServer server, ICommandSender sender, String[] args) throws CommandException
+    {
+        // args: <entity> <dim> [<x> <y> <z>]
```

Comment:
Can you make this also support the `/tp [target player] <destination player>` syntax?

Reply:
I'm trying to keep the command parsing simple, as it may well need to be redone for 1.13.
For now, you can always run `/forge setdim` followed by a `/tp` command.